iron-ajax.html

```diff
@@ -353,13 +353,15 @@
         contentType = 'application/x-www-form-urlencoded';
       }
       if (contentType) {
-        headers['Content-Type'] = contentType;
+        headers['content-type'] = contentType;
       }
       var header;
 
       if (this.headers instanceof Object) {
         for (header in this.headers) {
-          headers[header] = this.headers[header].toString();
+          // Normalize headers in lower case to make it easier for iron-request
+          // to reason about them.
+          headers[header.toLowerCase()] = this.headers[header].toString();
```

Comment:
My impulse would be to try it this way and wait to see if anyone complains. Almost all header-getting apis on the server are case insensitive, so I'm hoping that this won't be an issue.

The next best alternative I think is to use a more complicated data structure, something like:

headers[key.toLowerCase()] = {originalSpelling: key, value: value};

At present the code assumes that headers are spelled Like-This, and if someone sets a header differently we'll start doing a bunch of weird stuff, like setting multiple content-type headers, and mis-encoding body objects.